runtime/autoload/man.vim

```diff
@@ -1,6 +1,16 @@
-let s:man_tag_depth = 0
+" Ensure Vim is not recursively invoked (man-db does this)
+" by removing MANPAGER from the environment
+" More info here http://comments.gmane.org/gmane.editors.vim.devel/29085
+if &shell =~# 'fish$'
+  let s:man_cmd = 'set -e MANPAGER; man ^/dev/null'
+else
+  let s:man_cmd = 'unset MANPAGER; man 2>/dev/null '
+endif
```

Comment:
When `unlet $FOO` is implemented, we must remember to move this back to the ftplugin file...